cmd/fleet/serve.go

```diff
@@ -597,25 +600,80 @@ func trySendStatistics(ctx context.Context, ds fleet.Datastore, frequency time.D
 	return ds.RecordStatisticsSent(ctx)
 }
 
-func runCrons(ds fleet.Datastore, task *async.Task, logger kitlog.Logger, config config.FleetConfig, license *fleet.LicenseInfo, failingPoliciesSet fleet.FailingPolicySet) context.CancelFunc {
-	ctx, cancelBackground := context.WithCancel(context.Background())
-
-	ourIdentifier, err := server.GenerateRandomText(64)
+func runSchedules(
+	ctx context.Context,
+	ds fleet.Datastore,
+	logger kitlog.Logger,
+	config config.FleetConfig,
+	license *fleet.LicenseInfo,
+	failingPoliciesSet fleet.FailingPolicySet,
+) {
+	instanceID, err := server.GenerateRandomText(64)
 	if err != nil {
 		initFatal(errors.New("Error generating random instance identifier"), "")
 	}
+	appConfig, err := ds.AppConfig(ctx)
+	if err != nil {
+		level.Error(logger).Log("config", "couldn't read app config", "err", err)
+	}
+
+	webhooksLogger := kitlog.With(logger, "cron", "webhooks")
+	// webhooksInterval := appConfig.WebhookSettings.Interval.ValueOr(24 * time.Hour)
+	webhooksInterval := appConfig.WebhookSettings.Interval.ValueOr(30 * time.Second)
+	webhooks, err := schedule.New(ctx, "webhooks", instanceID, webhooksInterval, ds, webhooksLogger)
+	if err != nil {
+		fmt.Println("Error creating webhooks schedule: ", err)
+		sentry.CaptureException(err)
+
+	}
+	webhooks.SetConfigCheck(schedule.SetWebhooksConfigCheck(ctx, ds, webhooksLogger))
```

Comment:
It looks like you are setting the interval at which you should check for scheduled jobs based on the value of the of the webhook settings interval. Why is that?